GitCommands/ArgumentBuilder.cs

```diff
@@ -0,0 +1,260 @@
+                {
+                    builder.Add(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds the git argument syntax for members of the <see cref="ForcePushOptions"/> enum.
+        /// </summary>
+        /// <param name="builder">The <see cref="ArgumentBuilder"/> to add arguments to.</param>
+        /// <param name="option">The enum member to add to the builder.</param>
+        public static void Add(this ArgumentBuilder builder, ForcePushOptions option)
+        {
+            if (option == ForcePushOptions.Force)
+            {
+                builder.Add("-f");
+            }
+            else if (option == ForcePushOptions.ForceWithLease)
+            {
+                builder.Add("--force-with-lease");
+            }
+        }
+
+        /// <summary>
+        /// Adds the git argument syntax for members of the <see cref="UntrackedFilesMode"/> enum.
+        /// </summary>
+        /// <param name="builder">The <see cref="ArgumentBuilder"/> to add arguments to.</param>
+        /// <param name="mode">The enum member to add to the builder.</param>
+        public static void Add(this ArgumentBuilder builder, UntrackedFilesMode mode)
+        {
+            if (mode == UntrackedFilesMode.Default)
+            {
+                builder.Add("--untracked-files");
+            }
+            else if (mode == UntrackedFilesMode.No)
+            {
+                builder.Add("--untracked-files=no");
+            }
+            else if (mode == UntrackedFilesMode.Normal)
+            {
+                builder.Add("--untracked-files=normal");
+            }
+            else if (mode == UntrackedFilesMode.All)
+            {
+                builder.Add("--untracked-files=all");
+            }
+        }
+
+        /// <summary>
+        /// Adds the git argument syntax for members of the <see cref="IgnoreSubmodulesMode"/> enum.
+        /// </summary>
+        /// <param name="builder">The <see cref="ArgumentBuilder"/> to add arguments to.</param>
+        /// <param name="mode">The enum member to add to the builder.</param>
+        public static void Add(this ArgumentBuilder builder, IgnoreSubmodulesMode mode)
+        {
+            if (mode == IgnoreSubmodulesMode.Default)
+            {
+                builder.Add("--ignore-submodules");
+            }
+            else if (mode == IgnoreSubmodulesMode.None)
+            {
+                builder.Add("--ignore-submodules=none");
+            }
+            else if (mode == IgnoreSubmodulesMode.Untracked)
+            {
+                builder.Add("--ignore-submodules=untracked");
+            }
+            else if (mode == IgnoreSubmodulesMode.Dirty)
+            {
+                builder.Add("--ignore-submodules=dirty");
+            }
+            else if (mode == IgnoreSubmodulesMode.All)
+            {
+                builder.Add("--ignore-submodules=all");
+            }
```

Comment:
I am not particularly fond of so many `if/else` as I can't say I find it easier to read than a `switch` statement.
Besides `switch` typically has an edge over multiple `if/else` - https://stackoverflow.com/a/767840.

Beside in a `switch` it is easier to make default handling more obvious. And the default case can be leveraged for testing too:

```
switch (mode)
{
    // ...
    default: throw new NotSupportedException("developer you suck!");
}
```

And have a unit test to ensure that whenever the enum definition changes no exception is thrown. 
For example (pseudocode):

```
[Test]
public void Test_all_enum_values_mapped()
{
    var values = Enum.GetValues<EnumType>();
    foreach (var value in values)
    {
        ArgumentBuilderExtensions.Add(new ArgumentBuilder(), (EnumType)value).Should().NotThrow();
    }
}
```